tests/data_context/test_data_context_utils.py

```diff
@@ -286,6 +287,222 @@ def test_password_masker_mask_db_url(monkeypatch, tmp_path):
     assert PasswordMasker.mask_db_url("sqlite://", use_urlparse=True) == "sqlite://"
 
 
+def test_sanitize_config_raises_exception_with_bad_input(
+    basic_data_context_config,
+):
+    """Test case for PasswordMasker.sanitize_config"""
+
+    # expect that an Exception is raised if something other than a dict is passed
+    with pytest.raises(TypeError):
+        PasswordMasker.sanitize_config(basic_data_context_config)
+
+
+def test_sanitize_config_doesnt_change_config_without_datasources(
+    basic_data_context_config_dict,
+):
+    """Test case for PasswordMasker.sanitize_config"""
+
+    # expect no change without datasources
+    config_without_creds = PasswordMasker.sanitize_config(
+        basic_data_context_config_dict
+    )
+    assert config_without_creds == basic_data_context_config_dict
+
+
+def test_sanitize_config_masks_cloud_store_backend_access_tokens(
+    data_context_config_dict_with_cloud_backed_stores, ge_cloud_access_token
+):
+    """Test case for PasswordMasker.sanitize_config"""
+
+    # test that cloud store backend tokens have been properly masked
+    config_with_creds_in_stores = PasswordMasker.sanitize_config(
+        data_context_config_dict_with_cloud_backed_stores
+    )
+    for name, store_config in config_with_creds_in_stores["stores"].items():
+        try:
+            # check that the original token exists
+            assert (
+                data_context_config_dict_with_cloud_backed_stores["stores"][name][
+                    "store_backend"
+                ]["ge_cloud_credentials"]["access_token"]
+                == ge_cloud_access_token
+            )
+            # expect that the GE Cloud token has been obscured
+            assert (
+                store_config["store_backend"]["ge_cloud_credentials"]["access_token"]
+                != ge_cloud_access_token
+            )
+        except KeyError:
+            # a field in store_config["store_backend"]["ge_cloud_credentials"]["access_token"]
+            # doesn't exist, so we expect this config to be unchanged
+            assert (
+                store_config
+                == data_context_config_dict_with_cloud_backed_stores["stores"][name]
+            )
+
```

Comment:
I still am a bit iffy on this general pattern as it introduces some tricky conditional logic in a test, which should be as simple and straightforward as possible.

I won't block on this but I'm definitely of the opinion that the fewer asserts and overall branches, the easier the test is to read and maintain.

Reply:
I hear your point on the complexity of the test! I refactored the try/except to make it more straightforward to read. I personally find this approach to be preferable to a snapshot-style test since I see it as less fragile. From my perspective, this test shouldn't care about the rest of the contents of the config dict, only the fields under test.